src/Processors/Transforms/AggregatingTransform.h

```diff
@@ -70,6 +70,38 @@ struct ManyAggregatedData
         for (auto & mut : mutexes)
             mut = std::make_unique<std::mutex>();
     }
+
+    ~ManyAggregatedData()
+    {
+        try
+        {
+            // Aggregation states destruction may be very time-consuming.
+            // In the case of a query with LIMIT, most states won't be destroyed during conversion to blocks.
+            // Without the following code, they would be destroyed in the destructor of AggregatedDataVariants in the current thread (i.e. sequentially).
+            const auto pool = std::make_unique<ThreadPool>(variants.size());
+
+            for (auto && variant : variants)
+            {
+                // It doesn't make sense to spawn a thread if the variant is not going to actually destroy anything.
+                if (variant->aggregator)
+                {
+                    // variant is moved here and will be destroyed in the destructor of the lambda function.
+                    pool->trySchedule(
+                        [variant = std::move(variant), thread_group = CurrentThread::getGroup()]()
+                        {
+                            if (thread_group)
```

Comment:
And setThreadName?

Reply:
Do you mean smth more specific that the default "ThreadPool"? "AggregatesDestructionPool" / "AggregatesDestructionThread"?

Reply:
Yes (thread name is 15 byte max, so it will be AggregDestruct)